openedx/core/djangoapps/safe_sessions/middleware.py

```diff
@@ -768,6 +768,39 @@ def _get_encrypted_request_headers(request):
         return encrypt_for_log(str(request.headers), getattr(settings, 'SAFE_SESSIONS_DEBUG_PUBLIC_KEY', None))
 
 
+class EmailVerificationMiddleware(MiddlewareMixin):
+    """
+    Middleware responsible for invalidating user sessions
+    by detecting a mismatch between the user's email in
+    the session and request.user.email.
+
+    This middleware ensures that the sessions in other browsers
+    are invalidated (and remain validated in current browser)
+    when a user changes their email in one browser.
```

Comment:
Nit: It took me a moment to understand the parentheses, so I broke it out.
```suggestion
    This middleware ensures that the sessions in other browsers are invalidated 
    when a user changes their email in one browser. The active session in which
    the email change is made will remain valid.
```